pkg/cluster/internal/create/kubeadm-init.go

```diff
@@ -155,6 +152,32 @@ func runKubeadmInit(ec *execContext, configNode *NodeReplica) error {
 	return nil
 }
 
+// getHostPort retrives the random host where the API server/the control plane is exposed
+func getHostPort(ec *execContext, controlPlane *nodes.Node) (int, error) {
+	// in single control-plane cluster, the node exposing the API server to the outside world is the control-plane itself
+	node := controlPlane
+	port := kubeadm.APIServerPort
+
+	// in case an lb is defined, use the lb node instead
+	if ec.ExternalLoadBalancer() != nil {
+		loadBalancerHandle, ok := ec.NodeFor(ec.ExternalLoadBalancer())
+		if !ok {
+			return 0, errors.Errorf("unable to get the handle of load balancer node: %s", ec.ExternalLoadBalancer().Name)
+		}
+
+		node = loadBalancerHandle
+		port = haproxy.ControlPlanePort
+	}
+
+	hostPort, err := node.Ports(port)
+	if err != nil {
+		return 0, errors.Wrapf(err, "failed to get Port from node: %s", node.String())
+	}
+
+	log.Infof("kubernetes cluster is accessible from the host using port %d exposed by node %s", hostPort, node.String())
```

Comment:
tiny nit/suggestion: `s/is/should be/` -- we haven't actually verified that it's accessible, this kind of message can be frustrating when something isn't working 😅